File: TMessagesProj/src/main/java/tw/nekomimi/nekogram/ui/BottomBuilder.java
package tw.nekomimi.nekogram.ui;
import android.app.Activity;
import android.content.Context;
import android.content.DialogInterface;
import android.view.View;
import android.widget.LinearLayout;
import org.telegram.ui.ActionBar.BaseFragment;
import org.telegram.ui.ActionBar.BottomSheet;
import org.telegram.ui.Components.EditTextBoldCursor;
import org.telegram.ui.Cells.HeaderCell;
public class BottomBuilder {
    private BottomSheet.Builder builder;
    private LinearLayout rootView;
    public BottomBuilder(Context context) {
        builder = new BottomSheet.Builder(context);
        rootView = new LinearLayout(context);
        rootView.setOrientation(LinearLayout.VERTICAL);
    }
    public BottomBuilder(Activity activity, boolean needFocus) { this((Context) activity); }
    public BottomBuilder(Activity activity, boolean needFocus, int theme) { this((Context) activity); }
    public BottomBuilder(Context context, int theme) { this(context); }
    public BottomBuilder(Context context, boolean needFocus, int theme) { this(context); }
    public BottomBuilder(Context context, BaseFragment fragment) { this(context); }
    public BottomBuilder(Context context, boolean needFocus) { this(context); }
    public BottomSheet.Builder getBuilder() { return builder; }
    public LinearLayout getRootView() { return rootView; }
    public void addTitle(CharSequence title) {}
    public View addTitle(CharSequence title, String subtitle) { return null; }
    public View addTitle(String title, boolean big, String subtitle) { return null; }
    public HeaderCell addTitle(CharSequence title, boolean big) { return null; }
    public void addTitle(CharSequence title, CharSequence subtitle) {}
    // addItem variants
    public void addItem(String text, int icon, Runnable onClick) {}
    public void addItem(String text, int icon, boolean red, Runnable onClick) {}
    public void addItem(CharSequence text, int icon, boolean red, Runnable onClick) {}
    public void addItem(String text, int icon, kotlin.jvm.functions.Function1 onClick) {}
    public void addItem(String text, int icon, boolean red, kotlin.jvm.functions.Function1 onClick) {}
    // More add methods
    public void addItems(String[] texts, int[] icons, ItemClickListener listener) {}
    public void addItems(CharSequence[] texts, int[] icons, ItemClickListener listener) {}
    public void addRadioItem(String text, boolean checked, kotlin.jvm.functions.Function1 onClick) {}
    public void doRadioCheck(Object cell) {}
    public Object addButton(String text, boolean red, kotlin.jvm.functions.Function1 onClick) { return null; }
    public Object addButton(String text, kotlin.jvm.functions.Function1 onClick) { return null; }
    public Object addCheckItem(String text, boolean checked, boolean red, kotlin.jvm.functions.Function1 onClick) { return null; }
    public void addCancelItem() {}
    public void addCancelButton() {}
    public EditTextBoldCursor addEditText(String hint) { return null; }
    public void setOnPreDismissListener(DialogInterface.OnDismissListener listener) {}
    public BottomSheet create() { return builder.create(); }
    public BottomSheet show() { return builder.create(); }
    public void dismiss() {}
    public interface ItemClickListener { Object onClick(int which, CharSequence text, View cell); }
}

File: TMessagesProj/src/main/java/tw/nekomimi/nekogram/ui/BottomBuilder.kt
package tw.nekomimi.nekogram.ui

import android.content.Context
import android.content.DialogInterface
import android.widget.EditText
import android.widget.TextView
import org.telegram.ui.Cells.HeaderCell
import org.telegram.ui.Cells.RadioButtonCell
import org.telegram.ui.Cells.TextCell
import org.telegram.ui.Cells.TextCheckCell

/**
 * Stub: Bottom sheet builder - no-op implementation
 */
class BottomBuilder(val ctx: Context, val needFocus: Boolean = true, val bgColor: Int = 0) {
    @JvmField
    val bottomSheet = org.telegram.ui.ActionBar.BottomSheet(ctx, false)

    constructor(ctx: Context) : this(ctx, true)
    constructor(ctx: Context, needFocus: Boolean) : this(ctx, needFocus, 0)

    fun addTitle(title: CharSequence): HeaderCell = HeaderCell(ctx)
    fun addTitle(title: CharSequence, bigTitle: Boolean): HeaderCell = HeaderCell(ctx)
    fun addTitle(title: CharSequence, subTitle: CharSequence): HeaderCell = HeaderCell(ctx)
    fun addTitle(title: CharSequence, bigTitle: Boolean, subTitle: CharSequence?): HeaderCell = HeaderCell(ctx)

    fun addCheckItem(text: String, value: Boolean, listener: ((cell: TextCheckCell, isChecked: Boolean) -> Unit)?): TextCheckCell = TextCheckCell(ctx)
    fun addCheckItem(text: String, value: Boolean, switch: Boolean, valueText: String?): TextCheckCell = TextCheckCell(ctx)
    fun addCheckItem(text: String, value: Boolean, switch: Boolean, valueText: String?, listener: ((cell: TextCheckCell, isChecked: Boolean) -> Unit)?): TextCheckCell = TextCheckCell(ctx)
    fun addCheckItems(text: Array<String>, value: (Int) -> Boolean, listener: (index: Int, text: String, cell: TextCheckCell, isChecked: Boolean) -> Unit): List<TextCheckCell> = emptyList()
    fun addCheckItems(text: Array<String>, value: (Int) -> Boolean, switch: Boolean, listener: (index: Int, text: String, cell: TextCheckCell, isChecked: Boolean) -> Unit): List<TextCheckCell> = emptyList()
    fun addCheckItems(text: Array<String>, value: (Int) -> Boolean, switch: Boolean, valueText: ((Int) -> String)?, listener: (index: Int, text: String, cell: TextCheckCell, isChecked: Boolean) -> Unit): List<TextCheckCell> = emptyList()

    fun doRadioCheck(cell: RadioButtonCell) {}
    fun addRadioItem(text: String, value: Boolean, listener: (cell: RadioButtonCell) -> Unit): RadioButtonCell = RadioButtonCell(ctx, true)
    fun addRadioItem(text: String, value: Boolean, valueText: String?, listener: (cell: RadioButtonCell) -> Unit): RadioButtonCell = RadioButtonCell(ctx, true)
    fun addRadioItems(text: Array<String>, value: (Int, String) -> Boolean, listener: (index: Int, text: String, cell: RadioButtonCell) -> Unit): List<RadioButtonCell> = emptyList()
    fun addRadioItems(text: Array<String>, value: (Int, String) -> Boolean, valueText: ((Int, String) -> String)?, listener: (index: Int, text: String, cell: RadioButtonCell) -> Unit): List<RadioButtonCell> = emptyList()

    fun addCancelItem() {}
    fun addCancelButton() {}
    fun addCancelButton(left: Boolean) {}
    fun addOkButton(listener: ((TextView) -> Unit)) {}
    fun addOkButton(listener: ((TextView) -> Unit), noAutoDismiss: Boolean) {}
    fun addButton(text: String, listener: ((TextView) -> Unit)): TextView = TextView(ctx)
    fun addButton(text: String, noAutoDismiss: Boolean, listener: ((TextView) -> Unit)): TextView = TextView(ctx)
    fun addButton(text: String, noAutoDismiss: Boolean, left: Boolean, listener: ((TextView) -> Unit)): TextView = TextView(ctx)

    fun addItem(text: CharSequence, icon: Int, listener: ((cell: TextCell) -> Unit)?): TextCell = TextCell(ctx)
    fun addItem(text: CharSequence, icon: Int, red: Boolean, listener: ((cell: TextCell) -> Unit)?): TextCell = TextCell(ctx)
    fun addItems(text: Array<CharSequence?>, icon: IntArray?, listener: (index: Int, text: CharSequence, cell: TextCell) -> Unit): List<TextCell> = emptyList()

    fun addEditText(hintText: String? = null): EditText = EditText(ctx)

    fun setTitleMultipleLines(multilines: Boolean) {}
    fun create(): org.telegram.ui.ActionBar.BottomSheet = org.telegram.ui.ActionBar.BottomSheet(ctx, false)
    fun show(): android.app.Dialog {
        val bs = create()
        bs.show()
        return bs
    }
    fun dismiss() {}
    fun getBuilder(): BottomBuilder = this
    fun setOnPreDismissListener(onDismissListener: DialogInterface.OnDismissListener?): BottomBuilder = this
}
